ultravox/data/datasets.py

```diff
@@ -1142,8 +1144,12 @@ def __init__(
         self._static = static
 
         self._stop_strategy = stop_strategy
+        self._using_epochs = using_epochs
+        if not self._using_epochs:
+            weights = [int(getattr(ds, "weight", 1) * len(ds)) for ds in datasets]
```

Comment:
At this point, the term "weight" is incorrect. It seems like what you want is `data_multiplier`, not just a general notion of weight.

Another reason why, is that if you divide all "weights" by a constant, nobody expects anything to change, but in this case it does.